test/Adapter/PhpCodeTest.php

```diff
@@ -26,111 +25,71 @@ public function setUp()
         $this->adapter = new Serializer\Adapter\PhpCode();
     }
 
-    public function tearDown()
-    {
-        $this->adapter = null;
-    }
-
-    public function testSerializeString()
-    {
-        $value    = 'test';
-        $expected = "'test'";
-
-        $data = $this->adapter->serialize($value);
-        $this->assertEquals($expected, $data);
-    }
-
-    public function testSerializeFalse()
-    {
-        $value    = false;
-        $expected = 'false';
-
-        $data = $this->adapter->serialize($value);
-        $this->assertEquals($expected, $data);
-    }
-
-    public function testSerializeNull()
-    {
-        $value    = null;
-        $expected = 'NULL';
-
-        $data = $this->adapter->serialize($value);
-        $this->assertEquals($expected, $data);
-    }
-
-    public function testSerializeNumeric()
-    {
-        $value    = 100.12345;
-        $expected = '100.12345';
-
-        $data = $this->adapter->serialize($value);
-        $this->assertEquals($expected, $data);
-    }
-
+    /**
+     * Test when serializing a PHP object it matches the
+     * encode process
+     *
+     * Unserialize on PHP objects occur on Zend\Json\Encoder::encode
+     */
     public function testSerializeObject()
     {
-        $value    = new \stdClass();
```

Comment:
If we keep this test and make the changes suggested, from what I see the 'unserialize' will still fail.  Mainly due to __set_status: "PHP Fatal error:  Call to undefined method stdClass::__set_state()".  The testUnserializeObject is skipped because of this.

This was the main reason why I decided to re-work this adapter altogether.  What are your thoughts?